python/ray/serve/model_wrappers.py

```diff
@@ -248,6 +248,14 @@ async def predict(self, inp):
         """Perform inference directly without HTTP."""
         return await self.predict_impl(inp)
 
+    def reconfigure(
+        self,
+        checkpoint: Union[Checkpoint, str],
+        predictor_cls: Union[str, Type["Predictor"]],
+    ):
+        """Reconfigure Model from given Checkpoint"""
+        self.model = predictor_cls.from_checkpoint(checkpoint, predictor_cls)
```

Comment:
not quite. This call won't work because predictor_cls.from_checkpoint won't recognize the additional predictor_cls argument.

Please take a look at the init methods to see how are these arguments handled. It should be like:
```
        predictor_cls = _load_predictor_cls(predictor_cls)
        checkpoint = _load_checkpoint(checkpoint)
        self.model = predictor_cls.from_checkpoint(
            checkpoint, **predictor_from_checkpoint_kwargs
        )
```

Reply:
@simon-mo Thank you for your response. Since I started looking at Ray last week. I am still getting up to speed with the code base. It is so big. Lol!
With your directions I was able to find [Implementation](https://github.com/ray-project/ray/blob/master/java/serve/src/main/java/io/ray/serve/RayServeReplicaImpl.java#L287) of `reconfigure` which brought much clarity. 
For some reason earlier I used the `from_checkpoint` implementation of BatchPredictor as an example instead of looking at abstract in Predictor.

Reply:
This won't quite work. Because from_checkpoint is a classmethod, not a regular method. but you called it on self.model.from_checkpoint